lib/AST/ASTImporter.cpp

```diff
@@ -2239,19 +2312,20 @@ Decl *ASTNodeImporter::VisitRecordDecl(RecordDecl *D) {
       }
 
     } else {
-      D2 = RecordDecl::Create(Importer.getToContext(), D->getTagKind(), DC,
-                              StartLoc, Loc, Name.getAsIdentifierInfo(),
-                              PrevDecl);
+      bool AlreadyImported;
+      std::tie(D2, AlreadyImported) = CreateDecl<RecordDecl>(
+          D, Importer.getToContext(), D->getTagKind(), DC, StartLoc, Loc,
+          Name.getAsIdentifierInfo(), PrevDecl);
+      if (AlreadyImported)
+        return D2;
       D2->setLexicalDeclContext(LexicalDC);
       LexicalDC->addDeclInternal(D2);
     }
-    
+
     D2->setQualifierInfo(Importer.Import(D->getQualifierLoc()));
     if (D->isAnonymousStructOrUnion())
       D2->setAnonymousStructOrUnion(true);
   }
-  
-  Importer.Imported(D, D2);
```

Comment:
For same reason as at ObjC import, this `Imported` can not be removed (?). It is possible that `D2` is non-null when an existing record is found (that is probably not an imported one).

Reply:
True, very good catch.